### PR TITLE
Hotfix - #119 - workaround for stubbing internal and evaluated sources

### LINKED ISSUE #119: `PhpInternalSourceLocator` and `EvaledCodeSourceLocator` can only stub classes
As it stands, `PhpInternalSourceLocator` and `EvaledCodeSourceLocator` can only stub classes.

This is a limitation as per zendframework/zend-code#7

To fix it (for now), a simple solution is to apply a transformation to the output of `zendframework/zend-code`, but it is not sufficiently solid/stable.

### PR DESCRIPTION
**Pull request by Ocramius**
Should fix #119 for now.

### CODE REVIEW

File: src/SourceLocator/SourceStubber.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * @param ClassReflection $reflection
+     *
+     * @return string
+     */
+    public function __invoke(ClassReflection $reflection)
+    {
+        $stubCode   = ClassGenerator::fromReflection($reflection)->generate();
+        $interface  = $reflection->isInterface();
```
Comment (asgrim) [inline src/SourceLocator/SourceStubber.php]: Naming: imo, booleans should be `$isInterface` as communicates to the reader this should be `bool`

File: src/SourceLocator/SourceStubber.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        $trait      = $reflection->isTrait();
```
Comment (asgrim) [inline src/SourceLocator/SourceStubber.php]: as above, `$isTrait`

File: src/SourceLocator/SourceStubber.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * @param \PhpParser\Node[] $statements
+     * @param bool              $interfaceOrTrait (true => interface, false => trait)
+     * @param bool              $trait
+     *
+     * @return \PhpParser\Node[]
+     */
+    private function replaceNodesRecursively(array $statements, $interfaceOrTrait)
```
Comment (asgrim) [inline src/SourceLocator/SourceStubber.php]: Naming here - this is actually `$isInterface`

Comment (Ocramius) [inline src/SourceLocator/SourceStubber.php]: I meant to make it `$isInterfaceOfTrait`

**Pull request merged**